includes/embeds/class-amp-tumblr-embed-handler.php

```diff
@@ -6,56 +6,100 @@
  * @since 0.7
  */
 
+use AmpProject\Dom\Document;
+
 /**
  * Class AMP_Tumblr_Embed_Handler
  *
  * @internal
  */
 class AMP_Tumblr_Embed_Handler extends AMP_Base_Embed_Handler {
 
+	/**
+	 * Default width.
+	 *
+	 * Tumblr embeds for web have a fixed width of 540px.
+	 *
+	 * @link https://tumblr.zendesk.com/hc/en-us/articles/226261028-Embed-pro-tips
+	 * @var int
+	 */
+	protected $DEFAULT_WIDTH = 540;
+
+	/**
+	 * Base URL used for identifying embeds.
+	 *
+	 * @var string
+	 */
+	protected $base_embed_url = 'https://embed.tumblr.com/embed/post/';
+
 	/**
 	 * Register embed.
 	 */
 	public function register_embed() {
-		add_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10, 2 );
+		// Not implemented.
 	}
 
 	/**
 	 * Unregister embed.
 	 */
 	public function unregister_embed() {
-		remove_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10 );
+		// Not implemented.
 	}
 
 	/**
-	 * Filter oEmbed HTML for Tumblr to prepare it for AMP.
+	 * Sanitizes Tumblr raw embeds to make them AMP compatible.
 	 *
-	 * @param string $cache Cache for oEmbed.
-	 * @param string $url   Embed URL.
-	 * @return string Embed.
+	 * @param Document $dom DOM.
 	 */
-	public function filter_embed_oembed_html( $cache, $url ) {
-		$parsed_url = wp_parse_url( $url );
-		if ( false === strpos( $parsed_url['host'], 'tumblr.com' ) ) {
-			return $cache;
+	public function sanitize_raw_embeds( Document $dom ) {
+		$nodes = $dom->xpath->query( sprintf( '//div[ @class = "tumblr-post" and starts-with( @data-href, "%s" ) ]', $this->base_embed_url ) );
```

Comment:
Should we be concerned about there being other class names on the element other than `tumblr-post`?

Reply:
That's a valid concern to have, yes. We could either remove the `class` selector or check if the `class` attribute contains `tumblr-post`. I'd go with the former given that I think the query overall is specific enough for detecting Tumblr embeds.

Reply:
What about:

```suggestion
		$nodes = $dom->xpath->query( sprintf( '//div[ @class and contains( concat( ' ', normalize-space( @class ), ' ' ), ' tumblr-post ' ) and starts-with( @data-href, "%s" ) ]', $this->base_embed_url ) );
```

We're doing that elsewhere.

Reply:
Relying on `data-href` alone may have unintended side effects, as this attribute is used by other platforms as well. Maybe there'd be a "Click here to Tweet this URL!" button which has an arbitrary value for the `data-href` that could link off to Tumblr or any other URL.

Reply:
Good point. Let's go with your suggestion above then.